packages/eslint-plugin/src/rules/no-restricted-imports.ts

```diff
@@ -19,37 +19,95 @@ const baseRule = getESLintCoreRule('no-restricted-imports');
 export type Options = InferOptionsTypeFromRule<typeof baseRule>;
 export type MessageIds = InferMessageIdsTypeFromRule<typeof baseRule>;
 
+// In some versions of eslint, the base rule has a completely incompatible schema
+// This helper function is to safely try to get parts of the schema. If it's not
+// possible, we'll fallback to less strict checks.
+const tryAccess = <T>(getter: () => T, fallback: T): T => {
+  try {
+    return getter();
+  } catch {
+    return fallback;
+  }
+};
+
+const baseSchema = baseRule.meta.schema as {
```

Comment:
question: why did you switch back to basing off of the base rule's schema here?
Any reason or was it just to go back to de-duping things?

Reply:
No strong reason: just to dedupe things / keep things in sync.